crates/viewer/re_renderer/shader/world_grid.wgsl

```diff
@@ -0,0 +1,154 @@
+const ORIENTATION_YZ: u32 = 1;
+const ORIENTATION_XY: u32 = 2;
+
+struct VertexOutput {
+    @builtin(position)
+    position: vec4f,
+
+    @location(0)
+    scaled_world_plane_position: vec2f,
+};
+
+const PLANE_SCALE: f32 = 10000.0;
+
+// Spans a large quad where centered around the camera.
+//
+// This gives us the "canvas" to drawn the grid on.
+// Compared to a fullscreen pass, we get the z value (and thus early z testing) for free,
+// as well as never covering the screen above the horizon.
+@vertex
+fn main_vs(@builtin(vertex_index) v_idx: u32) -> VertexOutput {
+    var out: VertexOutput;
+
+    var plane_position = (vec2f(f32(v_idx / 2u), f32(v_idx % 2u)) * 2.0 - 1.0) * PLANE_SCALE;
+    var world_position: vec3f;
+    switch (config.orientation) {
+        case ORIENTATION_XZ: {
+            plane_position += frame.camera_position.xz;
+            world_position = vec3f(plane_position.x, 0.0, plane_position.y);
+        }
+        case ORIENTATION_YZ: {
+            plane_position += frame.camera_position.yz;
+            world_position = vec3f(0.0, plane_position.x, plane_position.y);
+        }
+        case ORIENTATION_XY: {
+            plane_position += frame.camera_position.xy;
+            world_position = vec3f(plane_position.x, plane_position.y, 0.0);
+        }
+        default: {
+            world_position = vec3f(0.0);
+        }
+    }
+
+    out.position = frame.projection_from_world * vec4f(world_position, 1.0);
+    out.scaled_world_plane_position = plane_position / config.spacing;
+
+    return out;
+}
+
+// Like smoothstep, but linear.
+// Used for antialiasing: smoothstep works as well but is subtly worse.
+fn linearstep(edge0: f32, edge1: f32, x: f32) -> f32 {
+    return saturate((x - edge0) / (edge1 - edge0));
+}
+fn linearstep2(edge0: vec2f, edge1: vec2f, x: vec2f) -> vec2f {
+    return saturate((x - edge0) / (edge1 - edge0));
+}
+
+// Distance to a grid line in x and y ranging from 0 to 1.
+fn calc_distance_to_grid_line(scaled_world_plane_position: vec2f) -> vec2f {
+    return 1.0 - abs(fract(scaled_world_plane_position) * 2.0 - 1.0);
+}
+
+@fragment
+fn main_fs(in: VertexOutput) -> @location(0) vec4f {
+    // Most basics are very well explained by Ben Golus here: https://bgolus.medium.com/the-best-darn-grid-shader-yet-727f9278b9d8
+    // We're not actually implementing the "pristine grid shader" which is a grid with world space thickness,
+    // but rather the pixel space grid, which is a lot simpler, but happens to be also described very well in this article.
+
+    // Distance to a grid line in x and y ranging from 0 to 1.
+    let distance_to_grid_line = calc_distance_to_grid_line(in.scaled_world_plane_position);
+
+    // Figure out the how wide the lines are in this "draw space".
+    let plane_unit_pixel_derivative = fwidthFine(in.scaled_world_plane_position);
+    let line_anti_alias = plane_unit_pixel_derivative;
+    let width_in_pixels = 1.0;//config.thickness_ui * frame.pixels_from_point;
```

Comment:
What's with the dead code?

Reply:
ohhh that's why my thickness control broke at some point 🤦

Reply:
haha 😆